### PR TITLE
feat(http): switch to new parser version

### PR DESCRIPTION
**Pull request by NTBBloodbath**
Hey, I recently updated the HTTP parser with a lot of breaking changes. This PR is intended to update the parser here so that it works correctly.

Some of the changes:
- Add `branch` field in the `install_info` table as the default branch name is `main` (I've had problems trying to install the parser if I don't set the branch manually as it tries to use `master` which does not exist, not sure if this is necessary)
- Update `http` revision in the lockfile
- Update queries
- Remove `generate_requires_npm` field in the `install_info` table, not required anymore

I have also added myself again as one of the maintainers, I now have free time again to help with that work. I hope there is no problem with that :)

### CODE REVIEW

**Comment by clason** [comment]
> Add branch field in the install_info table as the default branch name is main (I've had problems trying to install the parser if I don't set the branch manually as it tries to use master which does not exist, not sure if this is necessary)

That's not needed; you _do_ need to update the commit in the lockfile, though.

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -961,9 +961,9 @@ M.configs = {
     install_info = {
       url = 'https://github.com/rest-nvim/tree-sitter-http',
       files = { 'src/parser.c' },
-      generate_requires_npm = true,
+      branch = 'main',
```
Comment (clason) [inline lua/nvim-treesitter/parsers.lua:964]:
```suggestion
```

**Comment by clason** [comment]
PR should go to `master`; I'm rebasing `main` on top of all changes manually.

**Comment by clason** [comment]
Also, be aware of the new (mandatory!) capture names and query format (you can run `./scripts/format-queries.lua runtime/queries/http` to automate the latter).

**Comment by NTBBloodbath** [comment]
It seems like switching branches caused some weird conflicts in the parsers file. Do you want me to close this PR and make another one with the master branch as the base?

**Comment by clason** [comment]
That's probably easier, yes (you can use `git checkout --theirs` to avoid the massive conflict, but it's not a large PR and you can cherry-pick the important changes).

File: runtime/queries/http/highlights.scm [hunk 2]
```diff
@@ -1,48 +1,86 @@
...
+(external_body
+  file_path: (path) @text.uri)
 
 ; Comments
+
 (comment) @comment @spell
+
+; Errors
+
+(ERROR) @error
```
Comment (clason) [inline runtime/queries/http/highlights.scm:86]: And that _definitely_ needs to go.

Comment (NTBBloodbath) [reply runtime/queries/http/highlights.scm:86]: Noted, thanks for the heads-up!

**Comment by NTBBloodbath** [comment]
Done in #6316, I've also formatted the queries by using the script you mentioned above and updated the highlights query capture names as well

**Pull request closed without merging**